src/Conversion/KrnlToLLVM/ConvertKrnlToLLVM.cpp

```diff
@@ -531,27 +535,25 @@ bool extractConstantsToFile(ModuleOp &module, std::string filepath,
       alignment = op.getAlignment().value();
 
     // Padding if necessary.
-    if ((alignment > 0) && (packedConst.size() % alignment != 0)) {
+    if ((alignment > 0) && (totalConstSize % alignment != 0)) {
       uint64_t padSize =
-          ((uint64_t)(packedConst.size() / alignment) + 1) * alignment -
-          packedConst.size();
+          ((uint64_t)(totalConstSize / alignment) + 1) * alignment -
+          totalConstSize;
       SmallVector<char> pads(padSize, (char)0);
       packedConst.insert(packedConst.end(), pads.begin(), pads.end());
```

Comment:
We can reduce memory more by not using a padded buffer but two writes to a file: one for padding and one for rawData. For example.

```c
// Padding the current data for correct alignment.
uint64_t padSize = 0;
if ((alignment > 0) && (totalConstSize % alignment != 0)) {
 padSize = ((uint64_t)(totalConstSize / alignment) + 1) * alignment - totalConstSize;
 SmallVector<char> pads(padSize, (char)0);
 outfile.write(pads.data(), pads.size());
}

// Write the constant to a file.
uint64_t constSize = rawData.size();
outfile.write(rawData.data(), constSize);

// Update the total size and op's offset.
totalConstSize += padSize + constSize;
op.setOffsetAttr(b.getI64IntegerAttr(totalConstSize));
op.removeValueAttr();

```

Reply:
Smart, please do that.

Reply:
Thanks for your comments! I updated. 

`totalConstSize` in `op.setOffsetAttr(b.getI64IntegerAttr(totalConstSize))` is offset. So the `totalConstSize` before adding `constSize` should be used.

Reply:
Thanks!